airgun/entities/domain.py

```diff
@@ -0,0 +1,192 @@
+from navmazing import NavigateToSibling
+from wait_for import wait_for
+
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep, navigator
+from airgun.views.domain import (
+    DomainCreateView,
+    DomainListView,
+    DomainEditView
+)
+
+
+class DomainEntity(BaseEntity):
+    _DEFAULT_ORG = 'Default Organization'
+    _DEFAULT_LOC = 'Default Location'
+    _CREATE_SUCCESSFUL = 'Successfully created {}.'
+    _UPDATE_SUCCESSFUL = 'Successfully updated {}.'
+    _DELETE_SUCCESSFUL = 'Successfully deleted {}.'
+
+    @staticmethod
+    def _submit_domain_form(view, **kwargs):
```

Comment:
that whole method is redundant as test layer handle what is passed to entity and on entity level you only should say `view.fill()`, so necessary fields/widgets will be filled
refer to other entities implementation to get the idea

Reply:
My impression of this is the test layer does handle what's passed. This is a helper method to fill the fields in the view. My intention was for a tester to pass in the args they want to fill with something like this:

```python
session.domain.create(dns_domain='testdomain100', parameters=[{'name': 'myname', 'value': 'myvalue'}, {'name': 'myname2', 'value': 'myvalue2'}], locations=['Default Location'], organizations=['myorg1', 'myorg2'])
```

Reply:
OK I see, so your tests just pass in the raw view dict data